### PR TITLE
use cardinal of ordinal notation for ViewObj and Display

### PR DESCRIPTION
**Pull request by mohamed-barakat**

### CODE REVIEW

**Comment by mohamed-barakat** [comment]
The failed tests show warnings involving Julia stuff being undeclared.

**Comment by zickgraf** [comment]
> The failed tests show warnings involving Julia stuff being undeclared.

That's not a problem. The problem is an overfull hbox in the documentation:

> #W There are overfull boxes:
> Overfull \hbox (9.07803pt too wide) detected at line 2325

**Comment by zickgraf** [comment]
Please add two tests (viewing an isomorphism and displaying an empty set) for full code coverage.

**Comment by mohamed-barakat** [comment]
Now that the legacy `CategoryConstructor` is updated, this error seems to be an outdated docker container:

```
Error thrown by GAP: Error, BIND_GLOBAL: variable `IsTerminalCategoryWithMultipleObjects' must be unbound at /home/gap/.julia/artifacts/80196249c050204d1b65e3b9f8b41878914c2cb0/share/gap/lib/global.g:138 called from
BIND_GLOBAL( arg[1], CALL_FUNC_LIST( NewCategory, arg ) ); at /home/gap/.julia/artifacts/80196249c050204d1b65e3b9f8b41878914c2cb0/share/gap/lib/type.g:120 called from
<function "DeclareCategory">( <arguments> )
 called from read-eval loop at /home/gap/.julia/packages/CapAndHomalg/t535j/pkg/CategoryConstructor/gap/TerminalCategory.gd:27
```

**Comment by mohamed-barakat** [comment]
I will restart the test.

**Comment by zickgraf** [comment]
> Now that the legacy `CategoryConstructor` is updated, this error seems to be an outdated docker container:
> 
> ```
> Error thrown by GAP: Error, BIND_GLOBAL: variable `IsTerminalCategoryWithMultipleObjects' must be unbound at /home/gap/.julia/artifacts/80196249c050204d1b65e3b9f8b41878914c2cb0/share/gap/lib/global.g:138 called from
> BIND_GLOBAL( arg[1], CALL_FUNC_LIST( NewCategory, arg ) ); at /home/gap/.julia/artifacts/80196249c050204d1b65e3b9f8b41878914c2cb0/share/gap/lib/type.g:120 called from
> <function "DeclareCategory">( <arguments> )
>  called from read-eval loop at /home/gap/.julia/packages/CapAndHomalg/t535j/pkg/CategoryConstructor/gap/TerminalCategory.gd:27
> ```

That's another reason why we really have to get rid of suggested dependencies as much as possible: If `CategoryConstructor` would be a hard dependency, then we would clone it in the CI and everything would be fine. If it would not be a dependency at all, it would not be loaded, so again everything would be fine. But the suggested dependency picks up an outdated version...

> I will restart the test.

This is not sufficient, I have to rebuild the docker containers.

File: gap/SkeletalFinSetsForCAP.gi [hunk 1]
```diff
@@ -934,7 +934,67 @@ InstallMethod( ViewObj,
...
+InstallMethod( ViewObj,
+    "for a CAP map of skeletal finite sets",
+        [ IsSkeletalFiniteSetMap and IsIsomorphism ],
+        
+  function ( phi )
+    Print( "|", Length( Source( phi ) ), "| ⭇ |", Length( Range( phi ) ), "|" );
+end );
+
+##
+InstallMethod( DisplayString,
```
[the remark this answers is not in the document]
Comment (mohamed-barakat) [reply gap/SkeletalFinSetsForCAP.gi:977]: Changed `DisplayString` -> `PrintString`. And no, Display does not by default use `PrintString`.

**Comment by zickgraf** [comment]
Aaaaand the CI is broken again by the PR merged into MonoidalCategories.....

**Comment by zickgraf** [comment]
> > Aaaaand the CI is broken again by the PR merged into MonoidalCategories.....
> 
> there is a subsequent merge of CategoryConstructor :)

Yes, and a subsequent rebuild of gap-docker-base, gap-docker and gap-docker-master^^

File: gap/SkeletalFinSetsForCAP.gi (hunk 1 above)
[the remark this answers is not in the document]
Comment (zickgraf) [reply gap/SkeletalFinSetsForCAP.gi]: Ah, according to the documentation it does, but CAP is interfering.

**Comment by mohamed-barakat** [comment]
Many thanks 👍
